index.md

```diff
@@ -17,55 +20,79 @@ run `pipeline.wdl` using
+    "file": "File: A dbSNP VCF file",
+    "index": "File: The index (.tbi) file associated with the dbSNP VCF"
+  },
+  "pipeline.sampleConfigFiles": "Array[File]: Sample configuration files (see below)",
+  "pipeline.outputDir": "String: The path to the output directory",
+  "pipeline.reference": {
+    "fasta": "File: A reference fasta",
+    "fai": "File: The index associated with the reference fasta",
+    "dict": "File: The dict associated with the reference fatsa"
+  }
+}
+```
 
->All inputs have to be preceded by with `pipeline.`.
-Type is indicated according to the WDL data types: `File` should be indicators
-of file location (a string in JSON). Types ending in `?` indicate the input is
-optional, types ending in `+` indicate they require at least one element.
+Some additional inputs which may be of interest are:
+```JSON
+{
+  "pipeline.sample.Sample.library.Library.readgroup.numberChunks":
+    "Int (optional, default = 1): The number of chunks each fastq file should be split into for QC and alignment",
+  "pipeline.sample.Sample.library.Library.readgroup.Readgroup.mapping.platform":
+    "String? (optional, default = \"illumina\"): The sequencing platform used",
+  "pipeline.sample.Sample.library.Library.readgroup.Readgroup.mapping.AlignBwaMem.bwaMem.threads":
+    "Int (optional, default = 2): Number of threads used for alignment",
+  "pipeline.sample.Sample.createGvcf.Gvcf.scatterList.regions":
+    "File? (optional): Bed file with regions used for variantcalling",
+  "pipeline.sample.Sample.library.Library.bqsr.GatkPreprocess.scatterList.regions":
+    "File? (optional): Bed file with regions used for variantcalling",
+  "pipeline.genotyping.JointGenotyping.scatterList.regions":
+    "File? (optional): Bed file with regions used for variantcalling",
+  "pipeline.sample.Sample.library.Library.bqsr.scatterSize":
+    "Int (optional, default = 10000000): Size of scatter regions (see explanation of scattering below)",
+  "pipeline.genotyping.scatterSize":
+    "Int (optional, default = 10000000): Size of scatter regions (see explanation of scattering below)",
+  "pipeline.sample.Sample.createGvcf.scatterSize":
+    "Int (optional, default = 10000000): Size of scatter regions (see explanation of scattering below)"
+}
+```
 
 ### Sample configuration
 The sample configuration should be a YML file which adheres to the following
 structure:
 ```YML
-samples:
+samples: #Biological replicates
   <sample>:
-    libraries:
+    libraries: #Technical replicates
       <library>:
-        readgroups:
+        readgroups: #Sequencing lanes
           <readgroup>:
-              R1: <Path to first-end FastQ file.>
-              R1_md5: <MD5 checksum of first-end FastQ file.>
-              R2: <Path to second-end FastQ file.>
-              R2_md5: <MD5 checksum of second-end FastQ file.>
+              reads:
+                R1: <Path to first-end FastQ file.>
+                R1_md5: <MD5 checksum of first-end FastQ file.>
```

Comment:
first-end -> forward-end
second-end -> reverse-end

Reply:
I'm not sure that this is strictly true, doesn't the dUTP protocol for RNAseq cause R1 to be the reverse-end and R2 the forward-end?

Reply:
the general convention is use forward and reverse. Even in dUTP, the fastq (from cDNA) should follow this convention. But I am fine with first and second too since that won't be misleading.